### PR TITLE
reject invalid invite link acceptances

### PR DESCRIPTION
**Pull request by AMarcedone**
Need to add a couple more tests, but you can start taking a look

### CODE REVIEW

File: go/libkb/api_mock.go [hunk 1]
```diff
@@ -30,22 +30,125 @@ func (n *NullMockAPI) PostRaw(MetaContext, APIArg, string, io.Reader) (*APIRes,
...
+func (a *APIArgRecorder) PostDecode(mctx MetaContext, arg APIArg, arw APIResponseWrapper) error {
+	err := a.API.PostDecode(mctx, arg, arw)
+	a.Records = append(a.Records, APIRecord{Arg: arg, Method: MethodPostDecode, RespWrapper: arw, Err: err})
+	return err
+}
+func (a *APIArgRecorder) PostDecodeCtx(ctx context.Context, arg APIArg, arw APIResponseWrapper) error {
+	err := a.API.PostDecodeCtx(ctx, arg, arw)
+	a.Records = append(a.Records, APIRecord{Arg: arg, Method: MethodPostDecodeCtx, RespWrapper: arw, Err: err})
+	return err
+
```
Comment (zapu) [inline go/libkb/api_mock.go:141]: extra newline

File: go/teams/multiple_use_invite_test.go [hunk 2]
```diff
@@ -192,3 +212,197 @@ func TestSeitanHandleExceededInvite(t *testing.T) {
...
+	require.Contains(t, record.Err.Error(), "acceptance not found")
+
+	// now change the akey to something that cannot be b64 decoded
+	fakeMsg = origMsg.DeepCopy()
+	fakeMsg.Seitans[0].Akey = keybase1.SeitanAKey("*") + fakeMsg.Seitans[0].Akey[1:]
+	records = adminCallsHandleTeamSeitanAndReturnsRejectCalls(fakeMsg)
+	require.Len(t, records, 1, "one invite acceptance should be rejected")
+	record = records[0]
+	assertRejectInviteArgs(record, accepted.inviteID, uv.Uid, uv.EldestSeqno)
+	// We do not test the rejection error here, as team_rekeyd might or might
```
Comment (zapu) [inline go/teams/multiple_use_invite_test.go:308]: maybe change the underlying api to a mock that drops the requests, so you don't have to do this re-post every time

File: go/teams/multiple_use_invite_test.go [hunk 3]
```diff
@@ -192,3 +212,197 @@ func TestSeitanHandleExceededInvite(t *testing.T) {
...
+	require.NoError(t, err)
+	kbtest.Logout(tc)
+
+	admin, err := kbtest.CreateAndSignupFakeUser("team", tc.G)
+	require.NoError(t, err)
+
+	teamName, teamID := createTeam2(tc)
+	t.Logf("Created team %s", teamName)
+
+	// Add team invite link which expires in an hour.
```
Comment (zapu) [inline go/teams/multiple_use_invite_test.go:239]: the expTime is 10 minutes though

**Comment by AMarcedone** [comment]
Updated. PTAL

**Pull request merged**